internal/pkg/deploy/deploy.go

```diff
@@ -87,6 +95,82 @@ func NewStore(sessProvider SessionProvider, store ConfigStoreClient) (*Store, er
 	return s, nil
 }
 
+// Pipeline is a deployed pipeline.
+type Pipeline struct {
+	appName      string
+	resourceName string
+	isLegacy     bool
+}
+
+// NewPipeline initiates a new pipeline. Note that this function should only be used for unit tests.
+func NewPipeline(appName, resourceName string, isLegacy bool) Pipeline {
```

Comment:
nit: I can't decide how I feel about this, should we have the `Pipeline` fields public and avoid the constructor since `Pipeline` is just an aggregate data type?

I feel like a constructor makes sense if you're going to take interfaces as inputs